core/src/main/java/org/bitcoinj/wallet/CoinSelection.java

```diff
@@ -33,8 +34,18 @@ public class CoinSelection {
     public final Coin valueGathered;
     public final Collection<TransactionOutput> gathered;
 
+    public CoinSelection(Collection<TransactionOutput> gathered) {
+        this(sumOutputValues(gathered), gathered);
+    }
+
     public CoinSelection(Coin valueGathered, Collection<TransactionOutput> gathered) {
         this.valueGathered = valueGathered;
         this.gathered = gathered;
     }
+
+    private static Coin sumOutputValues(Collection<TransactionOutput> outputs) {
+        final AtomicLong sats = new AtomicLong(0);
+        outputs.forEach(output -> sats.addAndGet(output.getValue().value));
```

Comment:
This can be done better with a `Stream` IMO (no need for `AtomicLong` as a mutable accumulator). I'll post suggested code in a few minutes.

Reply:
Here's the code:
```
        return outputs.stream()
            .map(TransactionOutput::getValue)
            .reduce(Coin.ZERO, Coin::add);
```

Reply:
And note that `sats.addAndGet` probably doesn't use the safe add that we're using in `Coin::add`.

Reply:
Thanks, I merged your code via this PR. However I just noticed this approach also puts a higher load on the garbage collector by creating a `Coin` instance per stream element (if I'm reading this correctly). This might actually be more relevant than the increased runtime, compared to just adding to a long or AtomicLong. Hopefully still not relevant enough…

Reply:
If you look at PR #2412 it avoids the garbage collection hit and creates a single new `Coin` for the total:

```
    public static Coin sum(Collection<TransactionOutput> outputs) {
        return Coin.ofSat(sumSats(outputs));
    }

    public static long sumSats(Collection<TransactionOutput> outputs) {
        return outputs.stream()
                .mapToLong(output -> output.value)
                .reduce(0, LongMath::checkedAdd);
    }
```